url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/BenchmarkInheritanceCall.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2009-2013, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml;
+
+import org.ejml.data.D1Matrix64F;
+import org.ejml.data.DenseMatrix64F;
+import org.ejml.ops.RandomMatrices;
+
+import java.util.Random;
+
+
+/**
+ * Benchmark that tests to see if referring the parent of the class versus the actual class
+ * has any performance difference.  The function used internally is matrix multiplication in "ikj" order.
+ *
+ * @author Peter Abeles
+ */
+public class BenchmarkInheritanceCall {
+
+    public static void multParent( D1Matrix64F a , D1Matrix64F b , D1Matrix64F c )
+    {
+        double dataA[] = a.data;
+        double dataB[] = b.data;
+        double dataC[] = c.data;
+
+        double valA;
+        int indexCbase= 0;
+        int endOfKLoop = b.numRows*b.numCols;
+
+        for( int i = 0; i < a.numRows; i++ ) {
+            int indexA = i*a.numCols;
+
+            // need to assign dataC to a value initially
+            int indexB = 0;
+            int indexC = indexCbase;
+            int end = indexB + b.numCols;
+
+            valA = dataA[indexA++];
+
+            while( indexB < end ) {
+                dataC[indexC++] = valA*dataB[indexB++];
+            }
+
+            // now add to it
+            while( indexB != endOfKLoop ) { // k loop
+                indexC = indexCbase;
+                end = indexB + b.numCols;
+
+                valA = dataA[indexA++];
+
+                while( indexB < end ) { // j loop
+                    dataC[indexC++] += valA*dataB[indexB++];
+                }
+            }
+            indexCbase += c.numCols;
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 38:72, 112:146.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.